### DISCUSSION TITLE
Cannot understand how to convert cp2k output to deepmd data

### DISCUSSION THREAD

**Discussion by EYErbil** [deepmodeling/dpdata#680] [Q&A] [answered]
Hello, I have directly downloaded the git repo, and put my cp2k data into cp2k folder. (Attached)
![image](https://github.com/deepmodeling/dpdata/assets/112732881/432e35da-2a6c-4fbd-afc2-0d2df7886874)

But when I run the code with dpdata C:\Users\USER\Desktop\Her_Şey\Dpdata_deepmd\dpdata-master\dpdata\cp2k -i cp2k/output -o deepmd/npy -O data -n                                                                                                 
I get error : NotImplementedError: CP2KEnergyForceFormat doesn't support System.from
But the cp2k folder should have cell.py and output.py, why cant I convert? can anybody help? All of the output folders have : 
![image](https://github.com/deepmodeling/dpdata/assets/112732881/f2646d5f-d7b1-42ff-a5e2-77e337b814dc)
and I have tried to get only the files and not the folders, but it still does not work

**Comment by robinzyb** [comment]
please try to install the dpdata using `pip install dpdata`

**Reply by EYErbil** [reply]
I already did so

**Reply by robinzyb** [reply]
I don't get it. From the description, you downloaded the git repo and put your cp2k data into the dpdata/cp2k folder. Do I understand correctly?

**Reply by EYErbil** [reply]
My main question would be, using dpdata, how can I utilize my cp2k data for deepmd model training, I am kind of lost right now, should I try cp2kdata? Are dpdata and cp2kdata necessarily do the same thing implicitly? 
I have some md files, and single point files, right now, I am trying single_point ones, 
![image](https://github.com/deepmodeling/dpdata/assets/112732881/626f0eee-7603-4ca2-afef-5dd2f20d1c02)
and my single point files are like that.

**Reply by robinzyb** [reply]
cp2kdata provides the plugin for dpdata, since the intrinsic parser for cp2k in dpdata is out of date. But the intrinsic cp2k parser parser still works for some lower versions of cp2k, for example v7.1.
back to your main question, it seems the command you are using is not correct. if the dpdata has been installed using pip.
The command to convert to dpmd dataset would be 
```shell
 dpdata -i cp2k/output -o deepmd/npy -O data
``` 
instead of what you have written. 
Please make sure the dpdata installed correctly in the first place

**Reply by EYErbil** [reply]
--- You are parsing data using package Cp2kData ---                                                                                                                                                                                          Atom names are fake chemical symbols as you set in cp2k input.                                                                                                                                                                               Traceback (most recent call last):                                                                                                                                                                                                             File "/home/eerbil20/miniconda3/envs/deepmd/bin/dpdata", line 8, in <module>                                                                                                                                                                   sys.exit(dpdata_cli())                                                                                                                                                                                                                     File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/dpdata/cli.py", line 56, in dpdata_cli                                                                                                                                convert(**vars(parsed_args))                                                                                                                                                                                                               File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/dpdata/cli.py", line 96, in convert                                                                                                                                   s = LabeledSystem(from_file, fmt=from_format, type_map=type_map)                                                                                                                                                                           File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/dpdata/system.py", line 200, in __init__                                                                                                                              self.from_fmt(                                                                                                                                                                                                                             File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/dpdata/system.py", line 237, in from_fmt                                                                                                                              return self.from_fmt_obj(load_format(fmt), file_name, **kwargs)                                                                                                                                                                            File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/dpdata/system.py", line 1205, in from_fmt_obj                                                                                                                         data = fmtobj.from_labeled_system(file_name, **kwargs)                                                                                                                                                                                     File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/cp2kdata/dpdata_plugin.py", line 61, in from_labeled_system                                                                                                           data['forces'] = cp2k_e_f.atomic_forces_list * AU_TO_EV/AU_TO_ANG                                                                                                                                                                        TypeError: unsupported operand type(s) for *: 'NoneType' and 'float'

**Reply by robinzyb** [reply]
If the run type of cp2k is `energy_force`, have you printed the forces in the output ?
https://robinzyb.github.io/cp2kdata/docs/dpdata_plugin.html please check out the setups in `input.inp` of cp2k

**Reply by EYErbil** [reply]
thank you, since those are all pre-given output files, using another input file is impossible right now,since I cannot re-run the calculations. would you suggest any way out of this? 
Is there any way to print the required information to somewhere using cp2kdata for dpdata to use, and convert the files to deepmd usable files.

**Reply by robinzyb** [reply, marked as answer]
there is no way around. dpdata and cp2kdata are just simple file parsers. If the data are missing in the beginning, the codes cannot make something out of nothing.

**Reply by EYErbil** [reply]
(deepmd) eerbil20@DESKTOP-EH52IE3:/mnt/c/Users/USER/Desktop/Her_Şey/Denemetahta$ dpdata cp2k.log -i cp2k/aimd_output -o deepmd/npy -O data                                                                                                   --- You are parsing data using package Cp2kData ---                                                                                                                                                                                          Traceback (most recent call last):                                                                                                                                                                                                             File "/home/eerbil20/miniconda3/envs/deepmd/bin/dpdata", line 8, in <module>                                                                                                                                                                   sys.exit(dpdata_cli())                                                                                                                                                                                                                     File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/dpdata/cli.py", line 56, in dpdata_cli                                                                                                                                convert(**vars(parsed_args))                                                                                                                                                                                                               File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/dpdata/cli.py", line 96, in convert                                                                                                                                   s = LabeledSystem(from_file, fmt=from_format, type_map=type_map)                                                                                                                                                                           File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/dpdata/system.py", line 200, in __init__                                                                                                                              self.from_fmt(                                                                                                                                                                                                                             File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/dpdata/system.py", line 237, in from_fmt                                                                                                                              return self.from_fmt_obj(load_format(fmt), file_name, **kwargs)                                                                                                                                                                            File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/dpdata/system.py", line 1205, in from_fmt_obj                                                                                                                         data = fmtobj.from_labeled_system(file_name, **kwargs)                                                                                                                                                                                     File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/cp2kdata/dpdata_plugin.py", line 86, in from_labeled_system                                                                                                           cp2kmd = Cp2kOutput(output_file=cp2k_output_name,                                                                                                                                                                                          File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/cp2kdata/output.py", line 115, in __init__                                                                                                                            parse_run_type()                                                                                                                                                                                                                           File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/cp2kdata/output.py", line 377, in parse_md                                                                                                                            self.md_info = parse_md_info(self.filename)                                                                                                                                                                                                File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/cp2kdata/block_parser/header_info.py", line 161, in parse_md_info                                                                                                     md_info = regrep(                                                                                                                                                                                                                          File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/monty/re.py", line 49, in regrep                                                                                                                                      gen = reverse_readfile(filename) if reverse else zopen(filename, "rt")                                                                                                                                                                     File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/site-packages/monty/io.py", line 46, in zopen                                                                                                                                       _name, ext = os.path.splitext(filename)                                                                                                                                                                                                    File "/home/eerbil20/miniconda3/envs/deepmd/lib/python3.10/posixpath.py", line 118, in splitext                                                                                                                                                p = os.fspath(p)                                                                                                                                                                                                                         TypeError: expected str, bytes or os.PathLike object, not NoneType  

Okay, now I tried MD outputs, but I get this error
